db-engine/src/main/java/com/splicemachine/db/catalog/types/IndexDescriptorImpl.java

```diff
@@ -276,9 +274,16 @@ public boolean isUnique()
     /** @see IndexDescriptor#baseColumnPositions */
     public int[] baseColumnPositions()
     {
+        if (isInvalidIndexDescriptor()) {
+            throw new UnsupportedOperationException();
+        }
         return baseColumnPositions;
     }
 
+    public boolean isInvalidIndexDescriptor() {
+        return baseColumnPositions.length != baseColumnStoragePositions.length;
```

Comment:
Is this sufficient? Could they have the same length but position-wise not correct? Like column A has position 1 and storage position 2, and `baseColumnPositions[0] == 1` but `baseColumnStoragePositions[0] == 3`?

Reply:
This is not sufficient to guarantee the validity of the index descriptor. But it's sufficient to catch upgrade issues from pre 2022 to post 2022. I will rename the function to reflect that.